src/cmd/cli/command/compose.go

```diff
@@ -91,28 +91,25 @@ func makeComposeUpCmd() *cobra.Command {
 			loader := configureLoader(cmd)
 
 			ctx := cmd.Context()
-			provider, err := getProvider(ctx, loader)
-			if err != nil {
-				return err
-			}
-
 			project, loadErr := loader.LoadProject(ctx)
 			if loadErr != nil {
 				if nonInteractive {
 					return loadErr
 				}
 
-				term.Warn(loadErr)
-				project, err := CreateProjectForDebug(loader)
+				term.Error("Cannot load project:", loadErr)
+				project, err := createProjectForDebug(loader)
 				if err != nil {
 					return err
 				}
 
-				track.Evt("Debug Prompted", P("reason", err))
-				// Call the AI debug endpoint using the original command context (not the tailCtx which is canceled)
-				if nil == cli.InteractiveDebug(ctx, client, provider, "", project, nil, loadErr) {
-					return nil // don't show the defang hint if debugging was successful
-				}
+				track.Evt("Debug Prompted", P("loadErr", loadErr))
+				return cli.InteractiveDebugForLoadError(ctx, client, project, loadErr)
+			}
+
+			provider, err := getProvider(ctx, loader)
```

Comment:
Why is it better to get the provider after trying to load the project? My hunch is that the other way around fails more quickly in error scenarios

Reply:
Nevermind I see the answer in the description:

> Further avoids prompting the user for the cloud provider when we already know the project failed to load (it would show a load warning first).